integrations/opensearch/src/haystack_integrations/components/retrievers/opensearch/open_search_hybrid_retriever.py

```diff
@@ -0,0 +1,288 @@
+from haystack.components.joiners import DocumentJoiner
+from haystack.components.joiners.document_joiner import JoinMode
+from haystack.core.serialization import component_from_dict, import_class_by_name
+from haystack.document_stores.types import FilterPolicy
+from haystack.lazy_imports import LazyImport
+
+from haystack_integrations.components.retrievers.opensearch import OpenSearchBM25Retriever, OpenSearchEmbeddingRetriever
+from haystack_integrations.document_stores.opensearch import OpenSearchDocumentStore
+
+logger = logging.getLogger(__name__)
+
+# Use LazyImport to conditionally import haystack super_component, which is only available in newer versions
+with LazyImport(
+    "To use the OpenSearchHybridRetriever you need a more recent version of haystack. Run 'pip install haystack-ai>=2.13.0'"  # noqa: E501
+) as haystack_imports:
+    from haystack import super_component
+
+# Trigger an error message when the OpenSearchHybridRetriever is imported without haystack-ai>=2.13.0
+haystack_imports.check()
+
+
+@super_component
+class OpenSearchHybridRetriever:
+    """
+    A hybrid retriever that combines embedding-based and keyword-based retrieval from OpenSearch.
+
+    This component requires haystack-ai>=2.13.0 to work properly.
+    """
+
+    def __init__(
+        self,
+        document_store: OpenSearchDocumentStore,
+        *,
+        embedder: TextEmbedder = None,
+        # OpenSearchBM25Retriever
+        filters_bm25: Optional[Dict[str, Any]] = None,
+        fuzziness: Union[int, str] = "AUTO",
+        top_k_bm25: int = 10,
+        scale_score: bool = False,
+        all_terms_must_match: bool = False,
+        filter_policy_bm25: Union[str, FilterPolicy] = FilterPolicy.REPLACE,
+        custom_query_bm25: Optional[Dict[str, Any]] = None,
+        # OpenSearchEmbeddingRetriever
+        filters_embedding: Optional[Dict[str, Any]] = None,
+        top_k_embedding: int = 10,
+        filter_policy_embedding: Union[str, FilterPolicy] = FilterPolicy.REPLACE,
+        custom_query_embedding: Optional[Dict[str, Any]] = None,
+        # DocumentJoiner
+        join_mode: Union[str, JoinMode] = JoinMode.RECIPROCAL_RANK_FUSION,
+        weights: Optional[List[float]] = None,
+        top_k: Optional[int] = None,
+        sort_by_score: bool = True,
+        # extra kwargs
+        **kwargs,
+    ):
+        """
+        Initialize the OpenSearchHybridRetriever, a super component to retrieve documents from OpenSearch using
+        both embedding-based and keyword-based retrieval methods.
+
+        This super component is tied to a SentenceTransformersTextEmbedder.
+
+        We don't explicitly define all the init parameters of the components in the constructor, for each
+        of the components, since that would be around 20+ parameters. Instead, we define the most important ones
+        and pass the rest as kwargs. This is to keep the constructor clean and easy to read.
```

Comment:
Is this still true?

Reply:
yes, we are only exposing a few parameters for the retrieveres